service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -285,4 +296,55 @@ public String getRequiredGcpProject(UUID workspaceId) {
         .getGcpCloudContext()
         .map(GcpCloudContext::getGcpProjectId);
   }
+
+  /**
+   * Grant a user permission to impersonate their pet service account in a given workspace. Unlike
+   * other operations, this does not run in a flight because it only requires one write operation.
+   * This operation is idempotent.
+   *
+   * @return The email identifier of the user's pet SA in the given workspace.
+   */
+  public String enablePetServiceAccountImpersonation(
+      UUID workspaceId, AuthenticatedUserRequest userRequest) {
+    final String serviceAccountUserRole = "roles/iam.serviceAccountUser";
+    // Validate that the user is at least a writer in this workspace, as only writers can run
+    // workflows.
+    Workspace workspace =
+        validateWorkspaceAndAction(
+            userRequest, workspaceId, SamConstants.SAM_WORKSPACE_WRITE_ACTION);
+    stageService.assertMcWorkspace(workspace, "enablePet");
+
+    String userEmail =
+        SamService.rethrowIfSamInterrupted(
+            () -> samService.getRequestUserEmail(userRequest), "getRequestUserEmail");
```

Comment:
I validated this again, it turns out you're right about Sam always returning the user's email and I was calling the wrong endpoint before. Modified this to always ask Sam for the email, since I believe it's possible to get the pet SA email from the token (though I'm not positive).
Good idea on the test, I added that to the `EnablePet` test. It needs to be commented out for now, but I locally validated that it works as expected if we don't grant project level SA permissions.